src/strategy/StrategyV1.sol

```diff
@@ -0,0 +1,678 @@
+
+    /// @notice Address of the rewardToken Minter.
+    /// @dev Eg. CRV Minter.
+    address public immutable minter;
+
+    /// @notice Address of the governance.
+    address public governance;
+
+    /// @notice Address of the future governance contract.
+    address public futureGovernance;
+
+    /// @notice Interface for Stake DAO token Accumulator
+    address public accumulator;
+
+    /// @notice Curve DAO token Fee Distributor
+    address public feeDistributor;
+
+    /// @notice Reward Token for veToken holders.
+    address public feeRewardToken;
+
+    /// @notice Stake DAO SDT Distributor
+    address public SDTDistributor;
+
+    /// @notice Address of the factory.
+    address public factory;
+
+    /// @notice Address accruing protocol fees.
+    address public feeReceiver;
+
+    /// @notice Percentage of fees charged on `rewardToken` claimed.
+    uint256 public protocolFeesPercent;
+
+    /// @notice Amount of fees charged on `rewardToken` claimed
+    uint256 public feesAccrued;
+
+    /// @notice Percentage of fees charged on `rewardToken` claimed to incentivize claimers.
+    uint256 public claimIncentiveFee;
+
+    /// @notice Map vaults address -> is vault active
+    mapping(address => bool) public vaults;
+
+    /// @notice Map `_asset_`to corresponding liquidity gauge.
+    mapping(address => address) public gauges;
+
+    /// @notice Map liquidity gauge address -> gauge type (0,1,2,3).
+    mapping(address => uint256) public lGaugeType;
+
+    /// @notice Map native liquidity gauge to Stake DAO Reward Distributor.
+    mapping(address => address) public rewardDistributors;
+
+    /// @notice Map addresses allowed to interact with the `execute` function.
+    mapping(address => bool) public allowed;
+
+    ////////////////////////////////////////////////////////////////
+    /// --- EVENTS & ERRORS
+    ///////////////////////////////////////////////////////////////
+
+    /// @notice Event emitted when governance is changed.
+    /// @param newGovernance Address of the new governance.
+    event GovernanceChanged(address indexed newGovernance);
+
+    /// @notice Error emitted when input address is null
+    error ADDRESS_NULL();
+
+    /// @notice Error emitted when low level call failed
+    error LOW_LEVEL_CALL_FAILED();
+
+    /// @notice Error emitted when sum of fees is above 100%
+    error FEE_TOO_HIGH();
+
+    /// @notice Error emitted when auth failed
+    error GOVERNANCE();
+
+    /// @notice Error emitted when auth failed
+    error UNAUTHORIZED();
+
+    /// @notice Error emitted when auth failed
+    error GOVERNANCE_OR_FACTORY();
+
+    /// @notice Error emitted when trying to allow an EOA.
+    error NOT_CONTRACT();
+
+    //////////////////////////////////////////////////////
+    /// --- CONSTRUCTOR
+    //////////////////////////////////////////////////////
+
+    constructor(address owner, address _locker, address _veToken, address _rewardToken, address _minter) {
```

Comment:
You probably want some checks against `address(0)`, especially for `owner`, considering the `initialize` function.